tests/model_serving/model_server/inference_service_configuration/test_isvc_env_vars_updates.py

```diff
@@ -0,0 +1,88 @@
+import pytest
+
+from tests.model_serving.model_server.inference_service_configuration.constants import (
+    BASE_ISVC_CONFIG,
+    ISVC_ENV_VARS,
+)
+from tests.model_serving.model_server.inference_service_configuration.utils import (
+    verify_env_vars_in_isvc_pods,
+)
+from utilities.constants import KServeDeploymentType, RunTimeConfigs
+
+pytestmark = [pytest.mark.sanity, pytest.mark.usefixtures("valid_aws_config")]
+
+
+ISVC_ENV_VARS_CONFIG = {
+    "env-vars": ISVC_ENV_VARS,
+}
+RAW_DEPLOYMENT_ISVC_CONFIG = {
+    **BASE_ISVC_CONFIG,
+    **ISVC_ENV_VARS_CONFIG,
+    "deployment-mode": KServeDeploymentType.RAW_DEPLOYMENT,
+}
+SERVERLESS_DEPLOYMENT_ISVC_CONFIG = {
+    **BASE_ISVC_CONFIG,
+    **ISVC_ENV_VARS_CONFIG,
+    "deployment-mode": KServeDeploymentType.SERVERLESS,
+}
+
+
+@pytest.mark.rawdeployment
+@pytest.mark.parametrize(
+    "model_namespace, openvino_kserve_serving_runtime, ovms_kserve_inference_service",
+    [
+        pytest.param(
+            {"name": "raw-env-update"},
+            RunTimeConfigs.ONNX_OPSET13_RUNTIME_CONFIG,
+            RAW_DEPLOYMENT_ISVC_CONFIG,
+        ),
+        pytest.param(
+            {"name": "raw-multi-env-update"},
+            RunTimeConfigs.ONNX_OPSET13_RUNTIME_CONFIG,
+            {
+                **RAW_DEPLOYMENT_ISVC_CONFIG,
+                "min-replicas": 4,
+            },
+        ),
+    ],
+    indirect=True,
+)
+class TestRawISVCEnvVarsUpdates:
+    def test_raw_with_isvc_env_vars(self, ovms_kserve_inference_service):
+        """Test adding environment variables to the inference service"""
+        verify_env_vars_in_isvc_pods(isvc=ovms_kserve_inference_service, env_vars=ISVC_ENV_VARS, vars_exist=True)
+
+    def test_raw_remove_isvc_env_vars(self, removed_isvc_env_vars):
+        """Test removing environment variables from the inference service"""
```

Comment:
After removed_isvc_env_vars call, do we need to wait for the pods to reflect the change before the verify_env_vars_in_isvc_pods call?

Reply:
this is the test -> reflect the change
`removed_isvc_env_vars` wait for new pods so we know we are running with pods after the change